hera_cal/apply_cal.py

```diff
@@ -227,17 +246,44 @@ def apply_cal(data_infilename, data_outfilename, new_calibration, old_calibratio
         add_to_history += '\nOLD_CALFITS_HISTORY: ' + old_hc.history + '\n'
     else:
         old_gains, old_flags = None, None
-
+    hd = io.HERAData(data_infilename, filetype=filetype_in)
     add_to_history = version.history_string(add_to_history)
-
+    no_red_weights = redundant_weights is None
     # partial loading and writing using uvh5
     if nbl_per_load is not None:
         if not ((filetype_in == 'uvh5') and (filetype_out == 'uvh5')):
             raise NotImplementedError('Partial writing is not implemented for non-uvh5 I/O.')
-        hd = io.HERAData(data_infilename, filetype='uvh5')
         for attribute, value in kwargs.items():
             hd.__setattr__(attribute, value)
-        for data, data_flags, _ in hd.iterate_over_bls(Nbls=nbl_per_load):
+        all_reds = redcal.get_reds(hd.antpos, pols=hd.pols, bl_error_tol=bl_error_tol, include_autos=True)
```

Comment:
If `redundant_average` is false and `redundant_solution` is false, I'd suggest not running this line.

Reply:
I've moved this into an appropriate if statement.